category_theory/limits/preserves.lean

```diff
@@ -52,21 +52,48 @@ class preserves_colimit (K : J ⥤ C) (F : C ⥤ D) : Type (max u₁ u₂ v) :=
 (preserves : Π {c : cocone K}, is_colimit c → is_colimit (F.map_cocone c))
 
 @[class] def preserves_limits_of_shape (J : Type v) [small_category J] (F : C ⥤ D) : Type (max u₁ u₂ v) :=
-Π {K : J ⥤ C}, preserves_limit K F
+Π (K : J ⥤ C), preserves_limit K F
 @[class] def preserves_colimits_of_shape (J : Type v) [small_category J] (F : C ⥤ D) : Type (max u₁ u₂ v) :=
-Π {K : J ⥤ C}, preserves_colimit K F
+Π (K : J ⥤ C), preserves_colimit K F
 
 @[class] def preserves_limits (F : C ⥤ D) : Type (max u₁ u₂ (v+1)) :=
-Π {J : Type v} {𝒥 : small_category J}, by exactI preserves_limits_of_shape J F
+Π (J : Type v) [small_category J], by exactI preserves_limits_of_shape J F
```

Comment:
Changing the `small_category J` to square brackets will cause problems I think. At least for `has_limits`, I had mysterious problems in later files until I switched to using `{}`s throughout. Did you change these for a specific reason?

Reply:
I was having trouble getting those `subsingleton` instances right, and it seemed these binders were strange, so I changed them to what seemed natural. I've just switched them back, as now that the proofs for the subsingleton instances are better it doesn't seem to matter locally.

I'm happy leaving them as `{}` now, but perhaps better would be to change them to `()` and `[]`, and then change them if necessary later?

Reply:
OK great, that's what I figured.

One thing you can try already is making the similar change in `has_limits` or `has_colimits`. For some reason I don't understand, this causes a lot of errors in `types.lean`. So my conclusion is that using `{}` in this situation is somehow better, though it would be nice to know why...